generator.yaml

```diff
@@ -8,9 +14,25 @@ operations:
 resources:
   Table:
     fields:
+      AttributeDefinitions:
+        compare:
+          is_ignored: true
+      KeySchema:
+        compare:
+          is_immutable: true
```

Comment:
`is_immutable` doesn't ignore the field comparison in the generated delta function. The reason I added an `is_ignored: true` is that the Dynamodb control plane sorts the KeySchema array, and sometimes causes the controller to reconcile for ever since `reflect.DeepEqual` takes in consideration the order to elements. I added a custom `equalKeySchemaArrays` function which properly compares the Custom Resource `Spec.KeySchema` with the sorted one returned by the dynamodb control plane.

Reply:
@A-Hilaly `is_immutable` is a configuration on the FieldConfig, not the CompareConfig, so this is being ignored entirely...

Reply:
You're I brain lagged here. Addressed